src/components/LanguageOption.tsx

```diff
@@ -57,6 +66,14 @@ export default function LanguageOption(props: LanguageOptionProps) {
   const open = useOpenInNewPane(translation?.value?._ref, schemaType.name)
   const handleOpen = useCallback(() => open(), [open])
 
+  /* Once a translation has been created, reset the userHasClicked state to false
+   * so they can click on it to navigate to the translation. If a translation already
+   * existed when this component was mounted, this will have no effect. */
+  const hasTranslation = Boolean(translation)
+  useEffect(() => {
+    setUserHasClicked(false)
```

Comment:
Any reason to keep this here vs in a `finally()` callback on the transaction promise? I guess it may take a bit of time before `translation` comes back as truthy?

Reply:
yes, exactly.

While working I had originally put `setUserHasClicked(false)` within the promise - using not `finally()` but `then()` - but this resulted in a flash of content where:

- user clicks -> The button is disabled
- transaction resolves -> The button is no longer disabled but still showed the `+` icon
- `translation` becomes truthy -> The button is not disabled and shows the split-pane icon

This `useEffect` seemed more appropriate - we don't want the button to be clickable until the transaction has resolved **and** a translation is provided by the updated props.

Reply:
(Sorry for the slow reply, I was OOO all last week)

Reply:
Makes sense, thanks for clarifying!